### DISCUSSION TITLE
Intellij does not startup

### DISCUSSION THREAD

**Discussion by KianRolf** [devonfw/IDEasy#794] [Q&A] [answered]
I run ide intellij and it shows me the regular log messages with success but no Intellij Window shows up no matter how often I repeat the ide intellij command. What am I doing wrong or what could I do to make it work?

**Comment by hohwille** [comment, marked as answer]
For some reason IntelliJ starts itself as a kind of daemon. That means if you run intellij (e.g. on Windows `idea64.exe`), it checks if an instance of that process is already running. In such case it only sends a message to that process asking to open the according project with the given parameters.

However, this daemon approach is not implemented bullet-proof but has some bugs.
It may happen that an IntelliJ process is hanging on your computer.
You then have to open your task manager and find the IntelliJ process (on windows search for `idea64`) and kill it.
After no IntelliJ process is left and you run `ide intellij` again, it is working again.
Unfortunately there is no way for IDEasy to detect if that problem occurred.
This is only a bug on IntelliJ and all we can do is offer this workaround to you as a help.

There is also another problem of IntelliJ that we discovered:
When plugins are installed automatically, IntelliJ renders a popup dialog (looks like Swing) asking if you trust the plugin.
Unless this popup is confirmed the intellij process is blocked and the IntelliJ UI will not launch.
In some cases the popup stays in the background or even remains totally invisible.
Our expectation of IntelliJ would be that the IntelliJ UI would show up and inside there according popups will appear for such things.
We do understand that this would significantly complicate the boostrapping of IntelliJ since plugins shall not be loaded before they are trusted. And if they are loaded after the UI is open, things may need to change dynamically. However disabling and enabling plugins at runtime seems already supported but there are some edge-cases. Further, when we install a plugin via IntelliJ CLI why does IntelliJ not trust itself? This seems somehow inconsistent. Maybe IntelliJ also has some valid reason for this because they want the human end-user to be in control what to trust and want to prevent that an attacher or malware installs a malicious plugin without the user agreeing. All fine but as a result do not blame IDEasy for the resulting problems. There is nothing we can do about it.
See also https://github.com/devonfw/IDEasy/issues/1230
